tests/test_mock_http_client.py

```diff
@@ -0,0 +1,68 @@
+import pytest
```

Comment:
Acho que seria interessante ter um teste aqui que valide todos os métodos, só para que no futuro quando implementarmos algum método novo esse teste quebre para lembrarmos de criar o mock dele, o que vc acha ?


Pseudo código
```python 
methods = dir(nubank_client)

for method_name in methods:
  method = getattr(nubank_client, method_name) # Podemos filtrar os métodos privados aqui
  arg_count = method.__code__.co_argcount
  assert method(*range(arg_count))
```

Reply:
Eu criei novos testes e também deixei o mais automatico possível. Se alguem criar uma função no nubank.py ele deverá quebrar até termos um teste lá no test_mock_http_client.py.

Reply:
Eu tinha pensado algo diferente. Ao invés de forçar a criação de um teste pra cada método do cliente do Nubank, em um teste poderiamos testar que todos os métodos do Nubank funcionam com o MockHttpClient.

Por isso que no exemplo que eu deixei eu chamava todos os métodos o `Nubank` para ver se não ia levantar nenhuma exceção.

Digo isso pq os testes desse cliente são simplesmente asserts que só validam que o retorno não é falso.

Reply:
Fazer isso, vai gerar mudanças no código em outras áreas. Por exemplo a função `get_bill_details`, neste caso é passado um dicionário com a `bill `que deseja ser coletada mais informações.

Quando eu chamo usando o código que você passou, ou seja, genérico para todas as funções. Ele passa o `bill` como inteiro e, como ele espera um dicionario, acaba dando erro por outros motivos. Neste caso, melhorei o seu algoritmo para criar os parametros de acordo com a anotação dos parametros das funções.

Acredito que com isso, matamos o problema e realmente ficou mais simples de trabalhar.

Reply:
Hm, não tinha pensado nisso...